examples/benchmarks.py

```diff
@@ -362,18 +416,60 @@ def _compute_pytorch(model_names, dictionary, average_over, device, torchscript,
                             inference = model
                             inference(sequence)
 
-                        print("Going through model with sequence of shape", sequence.shape)
-                        runtimes = timeit.repeat(lambda: inference(sequence), repeat=average_over, number=3)
-                        average_time = sum(runtimes) / float(len(runtimes)) / 3.0
-                        dictionary[model_name]["results"][batch_size][slice_size] = average_time
+                        if not no_memory:
+                            # model.add_memory_hooks()  # Forward method tracing (only for PyTorch models)
+
+                            # Line by line memory tracing (all code in the module `transformers`) works for all models/arbitrary code
+                            trace = start_memory_tracing("transformers")
+                            inference(sequence)
+                            summary = stop_memory_tracing(trace)
+
+                            if verbose:
+                                print(
+                                    "\nLines by line memory consumption:\n"
+                                    + "\n".join(
+                                        f"{frame.filename}:{frame.line_number}: mem {cpu_gpu_mem.string}: {frame.line_text}"
+                                        for frame, _, _, cpu_gpu_mem in summary.sequential
+                                    )
+                                )
+                                print(
+                                    "\nLines with top memory consumption:\n"
+                                    + "\n".join(
+                                        f"=> {frame.filename}:{frame.line_number}: mem {cpu_gpu_mem.string}: {frame.line_text}"
+                                        for frame, _, _, cpu_gpu_mem in summary.cumulative[:6]
+                                    )
+                                )
+                                print(
+                                    "\nLines with lowest memory consumption:\n"
```

Comment:
is this useful output? are there negatives in `cumulative`?

Reply:
There are (in particular on my Ubuntu, there are some memory releases during execution which render the output noisier).

Also if you don't keep the output of the model, the last line is just a large release of memory to cancel all you've allocated haha.